aea/cli/local_registry_sync.py

```
@@ -0,0 +1,140 @@
+            continue
+        click.echo(f"Updating to {latest_public_id}.")
+        replace_package(str(package_id.package_type), latest_public_id, package_dir)
+
+
+def replace_package(
+    package_type: str, public_id: PublicId, package_dir: Union[Path, str]
+) -> None:
+    """
+    Download, extract and replace exists package.
+
+    :param package_type: str.
+    :param public_id: pacakge bulic id to download
+    :param: package_dir: target package dir
+
+    :return: None
+    """
+    with TemporaryDirectory() as tmp_dir:
+        new_package_dir = os.path.join(tmp_dir, public_id.name)
+        os.mkdir(new_package_dir)
+        fetch_package(
+            package_type, public_id=public_id, cwd=tmp_dir, dest=new_package_dir
+        )
+        shutil.rmtree(package_dir)
+        shutil.move(new_package_dir, package_dir)
+
+
+def get_package_latest_public_id(package_id: PackageId) -> PublicId:
+    """
+    Get package latest package id from the remote repo.
+
+    :param package_id: id of the package to check
+
+    :return: package id of the latest package in remote repo
+    """
+    package_meta = get_package_meta(
+        str(package_id.package_type), package_id.public_id.to_latest()
+    )
+    return PublicId.from_str(package_meta["public_id"])
+
+
+def enlist_packages(
+    base_dir: Union[Path, str]
+) -> Generator[Tuple[PackageId, Union[Path, str]], None, None]:
+    """
+    Generate list of the packages in local repo directory.
+
+    :param base_dir: path or str of the local repo.
+
+    :return: generator with Tuple of package_id and package diurectory.
+    """
+    for author in os.listdir(base_dir):
+        author_dir = os.path.join(base_dir, author)
+        if not os.path.isdir(author_dir):
+            continue  # pragma: nocover
+        for package_type_plural in os.listdir(author_dir):
+            if package_type_plural not in PACKAGES_DIRS:
+                continue  # pragma: nocover
```

We should raise an error when there are invalid directories.